### PR TITLE
Remove project reference in FileHDFio

### LINKED ISSUE #686: Rewrite hdf might destroy the entire hdf file
Currently, rewrite_hdf operated on a sub-group simply destroys the hdf file! I suggest to really do what the function is called and rewrite the entire hdf file instead of filtering by job_name or exclude groups/nodes.

```python
from pyiron_base import Project

project = Project('tbr')
root_hdf = project.create_hdf(project.path, 'projhdfio_full_2')

hdf = root_hdf.open("content")
hdf["answer"] = 42
print(hdf["answer"])

inner_hdf  = hdf.open('content')
inner_hdf['answer'] = 43
print(inner_hdf['answer'])

hdf.rewrite_hdf5('content')

hdf["answer"]
>>>
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-8-bbde202a3766> in <module>
     14 hdf.rewrite_hdf5('content')
     15 
---> 16 hdf["answer"]

/mnt/c/Users/Siemer/pyiron_git/pyiron_base/pyiron_base/generic/hdfio.py in __getitem__(self, item)
    164                             obj = self._convert_dtype_obj_array(obj)
    165                         return obj
--> 166                 raise ValueError(
    167                     "Unknown item: {} {} {}".format(item, self.file_name, self.h5_path)
    168                 )

ValueError: Unknown item: answer /mnt/c/Users/Siemer/pyiron/projects/tbr/projhdfio_full_2.h5 /projhdfio_full_2/content

```

### PR DESCRIPTION
**Pull request by niklassiemer**
Writing #673 I saw that we access `self.project` within the `FileHDFio`, however, this does not exist!

### CODE REVIEW

File: pyiron_base/generic/hdfio.py [hunk 1]
```diff
@@ -822,14 +822,12 @@ def rewrite_hdf5(
...
+        #_path = file_name.split("/")[-1]
+        #_path = ".".join(_path.split(".")[:-1])
         # path = '/'.join(p_lst[:-1])
-        new_file = _path + "_rewrite"
+        new_file = file_name + "_rewrite"
 
-        hdf_new = ProjectHDFio(
-            project=self.project, file_name=new_file, h5_path="/" + job_name
-        )
+        hdf_new = FileHDFio(file_name=new_file, h5_path="/" + job_name)
```
Comment (pmrv) [inline pyiron_base/generic/hdfio.py]: How did this work previously?  Or did we just always happen to call this from `ProjectHDFio`?

File: pyiron_base/generic/hdfio.py [hunk 2]
```diff
@@ -822,14 +822,12 @@ def rewrite_hdf5(
         if exclude_groups is None:
             exclude_groups = ["interactive"]
         file_name = self.file_name
-        _path = file_name.split("/")[-1]
-        _path = ".".join(_path.split(".")[:-1])
+        #_path = file_name.split("/")[-1]
+        #_path = ".".join(_path.split(".")[:-1])
         # path = '/'.join(p_lst[:-1])
```
Comment (pmrv) [inline pyiron_base/generic/hdfio.py]:
```suggestion
```

File: pyiron_base/generic/hdfio.py (hunk 1 above)
Comment (niklassiemer) [reply pyiron_base/generic/hdfio.py]: I assume so...

**Comment by niklassiemer** [comment]
Looking for use cases of the `rewrite_hdf5` function, I found  

https://github.com/pyiron/pyiron_atomistics/blob/56fa12d47b9b69b4260b2f28768babada00824c3/pyiron_atomistics/vasp/base.py#L436 and 
https://github.com/pyiron/pyiron_base/blob/c33df6924fca2003607b6697cf9f9a8731795c6f/pyiron_base/job/interactive.py#L303

For the former one, the exclude functionality seems to be used. However, I do not know if this is covered by tests. Thus @sudarsan-surendralal , could you have a quick look at this?

**Comment by sudarsan-surendralal** [comment]
> Looking for use cases of the `rewrite_hdf5` function, I found
> 
> https://github.com/pyiron/pyiron_atomistics/blob/56fa12d47b9b69b4260b2f28768babada00824c3/pyiron_atomistics/vasp/base.py#L436 and
> 
> https://github.com/pyiron/pyiron_base/blob/c33df6924fca2003607b6697cf9f9a8731795c6f/pyiron_base/job/interactive.py#L303
> 
> For the former one, the exclude functionality seems to be used. However, I do not know if this is covered by tests. Thus @sudarsan-surendralal , could you have a quick look at this?

Sorry I missed this. I'll have a look at this today.

**Comment by niklassiemer** [comment]
Silence old Data Structure warning on rewrite

**Comment by niklassiemer** [comment]
@sudarsan-surendralal did you manage to have a look? In my opinion, this is good to go if it is not interfering with the other use cases.

**Review by sudarsan-surendralal — APPROVED** [review]
LGTM! I tested this with some of my master jobs and they work fine!

**Pull request merged**